tock/hours/views.py

```diff
@@ -800,26 +802,30 @@ def last_timecard(self):
 
     def prefilled_formset(self):
         timecard = self.last_timecard()
-        project_ids, extra = [], 1
+        prefills = dict(TimecardPrefillData.objects.filter(
+            employee=self.request.user.user_data).values_list(
+            'project', 'hours'))
+        project_ids = []
+        extra = 1
         if timecard:
-            project_ids = set(
+            project_ids = [
                 tco.project_id for tco in
                 timecard.timecardobjects.all()
```

Comment:
if you're just extracting project_id, then this looks like a great use for values_list rather than iterating?

Reply:
I'll double check on that!